editor/editor_node.cpp

```diff
@@ -872,7 +872,11 @@ void EditorNode::_resources_changed(const Vector<String> &p_resources) {
 		}
 
 		if (!res->editor_can_reload_from_file()) {
-			continue;
+			Ref<Script> is_script = res;
+			// Scripts are reloaded via the script editor.
+			if (is_script.is_valid() && ScriptEditor::get_singleton()->get_open_scripts().has(res)) {
```

Comment:
```suggestion
			Ref<Script> scr = res;
			// Scripts are reloaded via the script editor.
			if (scr.is_null() || (scr.is_valid() && ScriptEditor::get_singleton()->get_open_scripts().has(scr))) {
```
The left part handles non-script resources, as otherwise they'd be always reloaded.